doc/fips/SONiC-OpenSSL-FIPS-140-3-deployment.md

```diff
@@ -0,0 +1,135 @@
+To enable the FIPS feature in the runtime as below, but it is required to restart the relative services, such as sshd, docker container telemetry, docker container restapi, etc.
+```
+echo 1 > /etc/fips/fips_enabled
+```
+To disable the FIPS feature.
+```
+echo 0 > /etc/fips/fips_enabled
+```
+
+When using the flag file, it is required to mount the file to the docker container, or set the flag inside of the container then it will take effect. For the SONiC devices with the new images after the runtime configuration feature implemented, we will mount the file for all the containers. For the SONiC devices with the old images, we need to do the additional configuration, to set the flag file for each of the docker containers.
+
+The sample script to enable FIPS for the devices with the old images:
+```
+mkdir -p /etc/fips
+echo 1 > /etc/fips/fips_enabled
+docker exec telemetry bash -c 'mkdir -p /etc/fips; echo 1 > /etc/fips/fips_enabled'
+docker exec restapi bash -c 'mkdir -p /etc/fips; echo 1 > /etc/fips/fips_enabled'
+docker restart telemetry restapi
+systemctl restart sshd
+```
+
+The script will be automatically triggred when the DB FIPS config change detected by the [hostcfgd](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-host-services-data/debian/sonic-host-services-data.hostcfgd.service), so it does not need to run it manually again in the new images.
+
+### FIPS enable Mode
+It is required to reboot the SONiC device when changing the FIPS mode from the mode none-enforce to the enforce mode, or from the mode enforce to the mode none-enforce.
+When installing the SONiC image at the first time, or upgrading the SONiC image, the default FIPS option is disabled by default (see [ENABLE_FIPS](https://github.com/sonic-net/sonic-buildimage/blob/6ba5b84d980983312f779ad65cfc8c90b9674707/rules/config#L292)). You can override the option in the build time to set ENABLE_FIPS=y, so it is not required for an additional reboot, after you install or upgrade the SONiC OS.
+
+When you plan to enable the enforce mode for all datacenters, you can build the SONiC with FIPS enabled by default, and set the configuration enforce=true for all devices.
+
+## SONiC FIPS State
+The SONiC FIPS state in the redis STATE_DB as below:
+| Key | Description |
+|:----:|---------------------|
+| enabled | The flag indicating whether the FIPS enabled, 1 enabled, others not enabled |
+| enforced | The flag indicating whether the FIPS enforced, 1 enforced, others not enforced |
+| enforced_next | The flag indicating whether the FIPS will be enforced in the next reboot, 1 enforced, others not enforced |
```

Comment:
>enforced_next [](http://example.com/codeflow?start=2&length=13)

enforced_next is more like ConfigDB enforce. It is a more like configuration than a state.

Reply:
Removed the enforced_next state, thanks.